django_feature_flip/helpers.py

```diff
@@ -1,22 +1,90 @@
 
-from django_feature_flip.models import Feature
+"""Main helper file of FeatureFlip."""
+
+from django_feature_flip.models import Group, Feature
 
 
 class FeatureFlip:
+    """Main class, singleton that should always be instanciated as FeatureFlip()."""
+
+    class FeatureFlipError(Exception):
+        """Default class for the FeatureFlip errors."""
+
+        def __init__(self, message):
+            """Initialize instance."""
+            self.message = message
+
     class __FeatureFlip:
+        """Initializer class for the singleton of FeatureFlip."""
+
         def __init__(self):
-            pass
+            """Initialize instance."""
+            self.groups = {}
 
         def __str__(self):
             return repr(self)
     instance = None
 
     def __init__(self):
+        """Initialize instance."""
         if not FeatureFlip.instance:
             FeatureFlip.instance = FeatureFlip.__FeatureFlip()
 
     def __getattr__(self, name):
+        """Get attribute."""
         return getattr(self.instance, name)
 
     def enabled(self, feature_name, actor=None):
-        return Feature.objects.get(name=feature_name).enabled(actor)
+        """Check if a feature is enabled for an actor."""
+        feature = get_feature_or_raise_error(feature_name)
+
+        if feature.totally_enabled:
+            return True
+        elif actor:
+            return self.feature_enabled_for_a_group_of_actor(feature, actor)
+        else:
+            return False
+
+    def register(self, group_name, func):
+        """Register the group of actors.
+
+        The group is defined as every actor that returns true to func(actor).
+        """
+        Group.objects.get_or_create(name=group_name)
+
+        self.groups[group_name] = func
+
+    def activate_group(self, feature_name, group_name):
+        """Turn on the group for a feature."""
+        feature = get_feature_or_raise_error(feature_name)
+
+        get_group_or_raise_error(group_name).features.add(feature)
```

Comment:
Doesn't this function need a return statement?

Reply:
What do you want to get returned?

Reply:
No I was just asking if it's an error or if nothing needs to be returned